packages/enzyme-adapter-react-16.1/src/ReactSixteenOneAdapter.js

```diff
@@ -295,12 +295,17 @@ class ReactSixteenOneAdapter extends EnzymeAdapter {
         return instance ? toTree(instance._reactInternalFiber).rendered : null;
       },
       simulateError(nodeHierarchy, rootNode, error) {
-        const { instance: catchingInstance } = nodeHierarchy
-          .find(x => x.instance && x.instance.componentDidCatch) || {};
+        const isErrorBoundary = ({ instance: elInstance, type }) => (
+          (elInstance && elInstance.componentDidCatch) || (type && type.getDerivedStateFromError)
+        );
+
+        const { instance: catchingInstance, type: catchingType } = nodeHierarchy
+          .find(isErrorBoundary) || {};
 
         simulateError(
           error,
           catchingInstance,
+          catchingType,
```

Comment:
so something here. we have to always be able to handle "old enzyme + new adapter", "new enzyme + old adapter", and "new enzyme + new adapter" - so adding an argument in the middle like this is a breaking change.

It'll have to be added at the end.

Reply:
OK, I'll move it to the end.